### PR TITLE
Fixes inconsistent  highlight issue in navbar

### LINKED ISSUE #5514: Inconsistent Page Highlighting in Campaign Navigation Bar
### What is happening?
There is an issue on the campaign page where the navigation bar does not consistently indicate the current page the user is viewing. The navigation menu includes items such as Home, Courses, Articles, Students, ORES, and Alerts, but only the "ORES" and "Alerts" pages are highlighted when the user is on those respective pages.

### To Reproduce

To replicate this issue, follow these steps:

- Navigate to the "Explore" menu in the navigation bar.
- Scroll down to the "New Campaign" table and select any row.
- On the subsequent page, observe that the "courses" tab on the campaign page is not marked as active or highlighted.
- Click on the other tabs, and you will notice that only the "ORES" and "Alerts" tabs are highlighted.

### Expected behaviour

The campaign navigation bar should consistently highlight the page the user is currently viewing by applying the "active" or selected state to the corresponding navigation item

### Screenshots
Screenshot 2023-10-19 at 7 42 43 AM

Screenshot 2023-10-19 at 7 44 51 AM

### System Information

 - OS: Mac Os
 - Browser Chrome
 - Version Sonoma 14.1

### PR DESCRIPTION
**Pull request by fillingtothemomo**
Fixes #5514 
## What this PR does
Used getLocation to match current path to that of different tabs and highlights the active tab in `app/assets/javascripts/components/common/campaign_navbar.jsx` and adds css in `app/assets/stylesheets/modules/_nav.styl`.

## Screenshots
Before:
[Screencast from 15-02-24 12:50:00 PM IST.webm](https://github.com/WikiEducationFoundation/WikiEduDashboard/assets/122573982/06966e76-d03c-483d-86e4-78cef32b17cb)

After:
[Screencast from 15-02-24 12:51:25 PM IST.webm](https://github.com/WikiEducationFoundation/WikiEduDashboard/assets/122573982/72b0d29c-c7b9-4122-8577-78ea90db15f1)

### CODE REVIEW

**Comment by ragesoss** [comment]
This looks like a much better way to implement it, thanks!

File: app/assets/stylesheets/modules/_nav.styl [hunk 1]
```diff
@@ -438,6 +438,8 @@ nav.profile-nav
           color #D3D3D3
         a:active
           background-color #3b4175 !important
+.active
```
[the remark this answers is not in the document]
Comment (ragesoss) [reply app/assets/stylesheets/modules/_nav.styl:441]: I note from the video that the styling is slightly different; only the underline is colored for the new implementation, but both the tab name and underline are colored for the already-working ones.

[the remark this answers is not in the document]
Comment (fillingtothemomo) [reply app/assets/stylesheets/modules/_nav.styl:441]: I apologize for taking so long, I am currently having midsems and also some issues with my ubuntu, I will make the required changes asap

[the remark this answers is not in the document]
Comment (fillingtothemomo) [reply app/assets/stylesheets/modules/_nav.styl]: [Screencast from 29-02-24 02:59:32 AM IST.webm](https://github.com/WikiEducationFoundation/WikiEduDashboard/assets/122573982/9626ebb9-0ae8-4d43-8bb7-fa0acbeb3a83)
I have made the required changes and added a video snippet.

**Comment by fillingtothemomo** [comment]
Please merge this PR @ragesoss .

**Comment by ragesoss** [comment]
There are some styling bugs with this. Different tabs have different heights for the navbar, and the alignment of the underline is not consistent.

The React tab is fine:
current

The server-rendered tab has a taller navbar and the underlines don't match up:
alignment

I'm also not sure why any CSS changes are necessary; ideally this should just use the same CSS that is already being used on the React tabs when they are active.

**Comment by fillingtothemomo** [comment]
@ragesoss the original css applies a.active to div of <a> tag whereas Navlink automatically applies a.active to its links so either we declare a separate css which is a bit buggy or we could use a useEffect hook to add active to <a>
[Screencast from 13-03-24 01:31:08 AM IST.webm](https://github.com/WikiEducationFoundation/WikiEduDashboard/assets/122573982/fc6a0b82-3bfd-4804-8625-87fe0acf350c)
 tags based on their links

**Comment by ragesoss** [comment]
Thanks for explaining. The useEffect hook seems reasonable; you should include a comment to the effect that the hook is only needed for the links that are not using NavLink and can be removed once the server-rendered pages are turned into React pages that use NavLink.

**Comment by fillingtothemomo** [comment]
> Thanks for explaining. The useEffect hook seems reasonable; you should include a comment to the effect that the hook is only needed for the links that are not using NavLink and can be removed once the server-rendered pages are turned into React pages that use NavLink.

Made the required changes.

**Comment by fillingtothemomo** [comment]
@ragesoss I believe this PR is ready to be merged unless there are anymore changes to be made. Please consider reviewing this https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/5687 as well. Thanks!

**Pull request merged**